pageserver/src/tenant/timeline.rs

```diff
@@ -187,7 +187,7 @@ pub struct Timeline {
     /// A lock to ensure that no layer of the timeline is removed concurrently by other tasks.
     /// This lock is acquired in [`Timeline::gc`], [`Timeline::compact`],
     /// and [`Tenant::delete_timeline`].
-    pub(super) layer_removal_cs: tokio::sync::Mutex<()>,
+    pub(super) layer_removal_cs: Arc<tokio::sync::Mutex<()>>,
```

Comment:
Lets describe why this needs to be an Arc. As I see this is needed because of spawn blocking as it requires for spawned closure to be 'static, so you cannot pass references to guards inside. This should go away with asyncification